Classes/Utility/DeeplBackendUtility.php

```diff
@@ -252,15 +250,10 @@ private static function getBackendUserAuthentication(): BackendUserAuthenticatio
         return $GLOBALS['BE_USER'];
     }
 
-    /**
-     * @return array{uid: int, title: string}|array<empty>
-     */
-    public static function detectCurrentPage(ProcessingInstruction $processingInstruction): array
+    public static function detectCurrentPage(ProcessingInstruction $processingInstruction): ?CurrentPage
     {
-        self::$currentPage = [];
-
         if ($processingInstruction->getProcessingTable() === 'pages') {
-            self::$currentPage = self::getPageRecord((int)$processingInstruction->getProcessingId());
+            self::$currentPage = CurrentPage::fromArray(self::getPageRecord((int)$processingInstruction->getProcessingId()));
```

Comment:
Use a private method on the same class instead of having a public static factory method on the DTO object.